src/ubuntu/16.04/Dockerfile

```diff
@@ -30,9 +30,7 @@ RUN apt-get install -y git \
             npm \
             tar \
             zip && \
-    apt-get clean && \
-    npm install -g azure-cli@0.9.20 && \
```

Comment:
Are there any compatibility concerns with installing the latest version (2.0.nn)?

Reply:
Hmm, potentially.  The problem is that 2.0.nn is the only version available:
```
root@612428bf000e:/# apt-cache madison azure-cli
 azure-cli | 2.0.78-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.77-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.76-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.75-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.74-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.73-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.72-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.71-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.70-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.69-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.68-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.67-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.66-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.65-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.64-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.63-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.62-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.61-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.60-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.59-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.58-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.56-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.55-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.54-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.53-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.52-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.51-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.50-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.49-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.47-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.46-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.45-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.44-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.43-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.42-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.41-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.38-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.37-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.34-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.33-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.32-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.31-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.30-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.29-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.28-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.27-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.26-2~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
 azure-cli | 2.0.26-1~xenial | https://packages.microsoft.com/repos/azure-cli xenial/main amd64 Packages
```

We could just leave the 16.04 Dockerfiles as is since they are still able to build with npm.  18.04 is another story though.  For that version, there's no other choice but to upgrade.

Reply:
The safe thing to do is to limit the impact (e.g. only change for 18.04).  It would be nice to know the potential impact to make a more educated decision as it would be nice to change across the board.

Reply:
Unfortunately, I can't find any release notes for those prior releases.  The oldest [release notes](https://docs.microsoft.com/en-us/cli/azure/release-notes-azure-cli?toc=%2Fcli%2Fazure%2Ftoc.json&bc=%2Fcli%2Fazure%2Fbreadcrumb%2Ftoc.json&view=azure-cli-latest) available start at 2.0.0, so it's difficult to know for sure what impact would be.